codespan-reporting/src/diagnostic.rs

```diff
@@ -80,7 +80,7 @@ impl<FileId> Label<FileId> {
 
 /// Represents a diagnostic message that can provide information like errors and
 /// warnings to the user.
-#[derive(Clone)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
```

Comment:
This seems a little large to derive Debug on ... can I ask your use case?

Reply:
The debug one I can actually remove. My main use case for the impls is salsa. Salsa requires that arguments for a query must implement hash, eq and partialEq

Reply:
This seems expensive to be in a salsa query, wouldn't it be faster to just recompute it instead of comparing all the strings to one another?